channel/pab.go

```diff
@@ -0,0 +1,239 @@
+	if err != nil {
+		return fmt.Errorf("failed to activate contract: %w", err)
+	}
+	p.contractInstanceID = response.Decode()
+	return nil
+}
+
+func (p *PAB) createSubscription(id channel.ID, isPerunSub bool) (*AdjudicatorSub, error) {
+	request := wire.MakeAdjudicatorSubscriptionActivationBody(id, p.acc.GetCardanoWalletID())
+	var response wire.ContractInstanceID
+	err := p.pabRemote.CallEndpoint(ActivateEndpoint, request, &response)
+	if err != nil {
+		return nil, fmt.Errorf("failed to activate subscription contract: %w", err)
+	}
+	subUrl := p.subscriptionUrlBase.JoinPath(response.Decode())
+	return newAdjudicatorSub(subUrl, id, isPerunSub)
+}
+
+// NewInternalSubscription creates a new adjudicator subscription for the given channel. The subscription will return
+// internal events. These are more specific to the Cardano implementation of the Perun contract and contain more
+// information. The internal events can not be used for anything go-perun related.
+// For this use NewPerunEventSubscription instead.
+func (p *PAB) NewInternalSubscription(id channel.ID) (*AdjudicatorSub, error) {
+	return p.createSubscription(id, false)
+}
+
+// NewPerunEventSubscription creates a new adjudicator subscription for the given channel. The subscription will return
+// perun events (generealized events compatible with the go-perun core).
+func (p *PAB) NewPerunEventSubscription(id channel.ID) (*AdjudicatorSub, error) {
+	return p.createSubscription(id, true)
+}
+
+// Start issues a request to the PAB to start the channel with the given parameters and initial state.
+func (p *PAB) Start(cid channel.ID, params types.ChannelParameters, state types.ChannelState) error {
+	if p.contractInstanceID == "" {
+		if err := p.activateContract(); err != nil {
+			return fmt.Errorf("failed to activate contract: %w", err)
+		}
+	}
+	request := wire.MakeOpenParams(cid, params, state)
+	err := p.pabRemote.CallEndpoint(fmt.Sprintf(StartEndpointFormat, p.contractInstanceID), request, nil)
+	if err != nil {
+		return fmt.Errorf("failed to call endpoint start: %w", err)
+	}
+	return nil
+}
+
+// Fund issues a request to the PAB to fund the channel with the given parameters.
+func (p *PAB) Fund(cid channel.ID, index channel.Index) error {
+	if p.contractInstanceID == "" {
+		if err := p.activateContract(); err != nil {
+			return fmt.Errorf("failed to activate contract: %w", err)
+		}
+	}
+	ct, err := p.GetChannelToken(cid)
+	if err != nil {
+		return fmt.Errorf("failed to fund channel: %w", err)
+	}
+	request := wire.MakeFundParams(cid, ct, uint16(index))
+	err = p.pabRemote.CallEndpoint(fmt.Sprintf(FundEndpointFormat, p.contractInstanceID), request, nil)
+	if err != nil {
+		return fmt.Errorf("failed to call endpoint fund: %w", err)
+	}
+	return nil
+}
+
+// Abort issues a request to the PAB to abort the channel. This only works on channels that
+// are not completely funded yet.
+func (p *PAB) Abort() {
+	//TODO
```

Comment:
A panic here is not possible, because of the way the `go-perun` framework calls this function? If it is posible, I would prefer a `panic` to force our hand (;